OctoKit/Event.swift

```diff
@@ -0,0 +1,147 @@
+import Foundation
+import RequestKit
+
+public extension Octokit {
+
+    /**
+     Fetches all recent Events on the server
+     - parameter session: RequestKitURLSession, defaults to NSURLSession.sharedSession()
+     - parameter completion: Callback for the outcome of the fetch.
+     */
+    @discardableResult
+    func allEvents(_ session: RequestKitURLSession = URLSession.shared,
+                  completion: @escaping (_ response: Response<[Event]>) -> Void) -> URLSessionDataTaskProtocol? {
+
+        let router = EventRouter.allEvents(self.configuration)
+        return router.load(session, expectedResultType: [Event].self) { events, error in
+            if let error = error {
+                completion(Response.failure(error))
+            } else {
+                if let events = events {
+                    completion(Response.success(events))
+                }
+            }
+        }
+    }
+
+    /**
+     Fetches all Received Events
+     - parameter session: RequestKitURLSession, defaults to NSURLSession.sharedSession()
+     - parameter completion: Callback for the outcome of the fetch.
+     */
+    @discardableResult
+    func myReceivedEvents(_ session: RequestKitURLSession = URLSession.shared,
+                   completion: @escaping (_ response: Response<[Event]>) -> Void) -> URLSessionDataTaskProtocol? {
+
+        let router = EventRouter.myReceivedEvents(configuration, "j0r010l")
+        return router.load(session, expectedResultType: [Event].self) { events, error in
+            if let error = error {
+                completion(Response.failure(error))
+            } else {
+                if let events = events {
+                    completion(Response.success(events))
+                }
+            }
+        }
+    }
+
+    /**
+     Fetches all Created Events
+     - parameter session: RequestKitURLSession, defaults to NSURLSession.sharedSession()
+     - parameter completion: Callback for the outcome of the fetch.
+     */
+    @discardableResult
+    func myCreatedEvents(_ session: RequestKitURLSession = URLSession.shared,
+                          completion: @escaping (_ response: Response<[Event]>) -> Void) -> URLSessionDataTaskProtocol? {
+
+        let router = EventRouter.myCreatedEvents(configuration, "j0r010l")
+        return router.load(session, expectedResultType: [Event].self) { events, error in
+            if let error = error {
+                completion(Response.failure(error))
+            } else {
+                if let events = events {
+                    completion(Response.success(events))
+                }
+            }
+        }
+    }
+}
+
+open class Event: Codable {
+    open var id: String
+    open var type: String
+    open var `public`: Bool
+    // open var payload: Any // TODO: what is this thing
```

Comment:
Payload is a dynamic thing, if you open a PR then it'll be a Pull request. 
you can tell which type of payload you have by looking at the `type` variable.

Looks like there is some custom `Codable` Implementation in my future.